Mathlib/Geometry/Synthetic/Avigad/Axioms.lean

```diff
@@ -0,0 +1,237 @@
+/--Lengths are nonnegative-/
+length_nonneg : ∀ a b, 0 ≤ length a b
+/--Areas are nonnegative-/
+area_nonneg : ∀ a b c, 0 ≤ area a b c
+/--Degenerate areas are zero-/
+degenerate_area : ∀ a b, area a a b = 0
+/--Area is completely symmetric-/
+area_invariant_cycle : ∀ a b c, area a b c = area c a b
+/--Area is completely symmetric-/
+area_invariant_flip : ∀ a b c, area a b c = area a c b
+/--If SSS is satisfied then triangles have equal area-/
+area_eq_of_SSS : ∀ {a b c a1 b1 c1}, length a b = length a1 b1 →
+  length a c = length a1 c1 → length b c = length b1 c1 → area a b c = area a1 b1 c1
+/--Given betweeness the lengths of segments add as expected-/
+length_sum_of_B : ∀ {a b c}, B a b c → length a b + length b c = length a c
+/--Points on a circle have the same distance from the radius-/
+on_circle_iff_length_eq : ∀ {a b c α}, CenterCircle a α → OnCircle b α →
+  (length a b = length a c ↔ OnCircle c α)
+/--A point on a circle has a greater distance from the center than a point inside the circle-/
+in_circle_iff_length_lt : ∀ {a b c α}, CenterCircle a α → OnCircle b α →
+  (length a c < length a b ↔ InCircle c α)
+/--One kind of degenerate angle is zero-/
+angle_zero_iff_online : ∀ {a b c L}, a ≠ b → a ≠ c → OnLine a L → OnLine b L →
+  (OnLine c L ∧ ¬B b a c ↔ angle b a c = 0)
+/--Conditions for a split angle to add as expected-/
+angle_add_iff_sameside : ∀ {a b c d L M}, a ≠ b → a ≠ c → OnLine a L →
+  OnLine b L → OnLine a M → OnLine c M → ¬OnLine d M → ¬OnLine d L → L ≠ M →
+  (angle b a c = angle b a d + angle d a c ↔ SameSide b d M ∧ SameSide c d L)
+/--Betweeness forces equal angles across the middle to be right angles-/
+angle_eq_iff_rightangle : ∀ {a b c d L}, OnLine a L → OnLine b L → ¬OnLine d L → B a c b →
+  (angle a c d = angle d c b ↔ angle a c d = rightangle)
+/--A condition to extend angles in a predictable way-/
+angle_extension : ∀ {a b c a1 b1 c1 L M}, b ≠ a → b1 ≠ a → c ≠ a → c1 ≠ a → OnLine a L →
+  OnLine b L → OnLine b1 L → OnLine a M → OnLine c M → OnLine c1 M → ¬B b a b1 → ¬B c a c1 →
+  angle b a c = angle b1 a1 c1
+/--The unparallel postulate-/
+unparallel_postulate : ∀ {a b c d L M N}, b ≠ c → OnLine a L → OnLine b L →
+  OnLine b M → OnLine c M → OnLine c N → OnLine d N →  SameSide a d M → angle a b c +
+  angle b c d < 2 * rightangle → ∃ e, OnLine e L ∧ OnLine e N ∧ SameSide e a M
+/--Areas of degenerate triangles equal zero-/
+area_zero_iff_online : ∀ {a b c L}, a ≠ b → OnLine a L → OnLine b L → (area a b c = 0 ↔ OnLine c L)
+/--Areas adding on a triangle given a betweeness condition-/
+area_add_iff_B : ∀ {a b c d L}, a ≠ b → b ≠ c → c ≠ a → OnLine a L →
+  OnLine b L → OnLine c L → ¬OnLine d L → (B a b c ↔ area d a b + area d c b = area d a c)
+/--SAS is equivalent to SSS-/
+SAS_iff_SSS : ∀ {a b c d e f}, length a b = length d e → length a c = length d f →
+  (angle b a c = angle e d f ↔ length b c = length e f)
+
+variable [i : IncidenceGeometry]
+open IncidenceGeometry
```

Comment:
I think you probably want
```suggestion
namespace IncidenceGeometry
```

Reply:
Ok! Just out of curiosity, why do we want to do this?  I changed it in the `Axioms` file and in `EuclidBookI`, but was not able to do it for `Tactics` since the `elab` expressions started to break?

Reply:
Again, for now this advice is only for this axioms file. It means that the axioms end up in the same namespace as the derived definitions.

Reply:
done